web_extension/content/scripts/new_tab.js

```diff
@@ -124,10 +138,16 @@ function update_date(now) {
   document.querySelector('date').textContent = date
 }
 
-async function fetch_config(){
-  window.config = await browser.storage.sync.get(default_options())
+async function fetchConfig() {
+  return options.read()
 }
 
-fetch_config()
-fetch_image()
-setInterval(tick, 100)
+const options = new Options()
+const display_options = options.display
+
+fetchConfig().then(() => {
+  document.querySelector('version').textContent = Version.number
+
+  fetchImage()
+  setInterval(tick, 100)
```

Comment:
This seems like it's a lot of work to be doing every 100ms, could you have a thing that updates settings only on a message from the background script which is messaged from the options page, and only have `tick` update the clock instead?

Reply:
The storage api has an event, supposedly, but I couldn’t get it to fire. Messaging is for plebeians